plugins/chat/assets/javascripts/discourse/components/chat/thread-list.js

```diff
@@ -1,24 +1,25 @@
 import Component from "@glimmer/component";
 import { bind } from "discourse-common/utils/decorators";
-import { tracked } from "@glimmer/tracking";
+import { cached } from "@glimmer/tracking";
 import { action } from "@ember/object";
 import { inject as service } from "@ember/service";
 
 export default class ChatThreadList extends Component {
   @service chat;
+  @service chatApi;
   @service messageBus;
+  @service chatTrackingStateManager;
 
-  @tracked loading = true;
+  get threadsManager() {
+    return this.args.channel.threadsManager;
+  }
 
   // NOTE: This replicates sort logic from the server. We need this because
   // the thread unread count + last reply date + time update when new messages
   // are sent to the thread, and we want the list to react in realtime to this.
   get sortedThreads() {
-    if (!this.args.channel.threadsManager.threads) {
-      return [];
-    }
-
-    return this.args.channel.threadsManager.threads
+    return this.threadsManager.threads
```

Comment:
Im quite mixed on this being a good solution, but I understand having a "standalone" array of threads comes with all sort of complexity to handle tracking, new threads, deleted OM, restored OM,...

It somehow makes me understand what kind of pattern we need here, some kind of abstraction on top of message bus in chat pane subscription where any class can subscribe to the same events but the final result of it:
- onNewThread
- onNewChannel
- onRestoredMessage
- onDeletedMessage
- ...

This would make this much easier and probably cleaner.

Reply:
Moreover, I can imagine edge case where the query is only loading threads which were already loaded and as a result had no new item in the list making scroll impossible.

Reply:
> It somehow makes me understand what kind of pattern we need here, some kind of abstraction on top of message bus in chat pane subscription where any class can subscribe to the same events but the final result of it

Yes this is what I was getting at back in https://github.com/discourse/discourse/pull/22402#discussion_r1251383720 . I already think there is some pain around these threads just being tracked in one place -- as you say there can be a load issue when querying threads when we already have them cached locally ,and I also think a similar issue is causing that issue where some threads double up in the list.

Maybe it would just be easier to have a separate thread store for the channel, which just handles showing the unread thread counter in the header and the thread indicator / last reply updates, and then a separate thread store that is only used for the list and when you open individual threads. Might make things a bit easier to reason about, but yes the subscription abstraction will be needed first.